### PR TITLE
Refactor some code after #15254

### PR DESCRIPTION
**Pull request by afc163**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [x] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

Improve #15254

### 💡 Solution

1. How to fix the problem, and list final API implementation and usage sample if that is an new feature.

2. GIF or snapshot should be provided if includes UI/interactive modification.

### 📝 Changelog description

> Describe changes from user side, and list all potential break changes or other risks.

1. English description

2. Chinese description (optional)

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
This pull request **fixes 1 alert** when merging da6662562bc632470c2a99fbbcfbc327f64d43af into 120ea3a38d356f36fa3643e768332936b8be04ec - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-80f5c764d38648701473f0142ed390da351be211)

**fixed alerts:**

* 1 for Potentially inconsistent state update

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: components/badge/ScrollNumber.tsx [hunk 1]
```diff
@@ -63,9 +62,9 @@ class ScrollNumber extends Component<ScrollNumberProps, ScrollNumberState> {
       return 10 + num;
     }
     const currentDigit = getNumberArray(this.state.count)[i];
-    const lastDigit = getNumberArray(lastCount)[i];
+    const lastDigit = getNumberArray(this.lastCount)[i];
     // 同方向则在同一侧切换数字
-    if (this.state.count! > lastCount) {
+    if (this.state.count! > this.lastCount!) {
```
Comment (zy410419243) [inline components/badge/ScrollNumber.tsx]: 很好奇这个感叹号是干嘛用的

Comment (afc163) [reply components/badge/ScrollNumber.tsx]: https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

断言，就是手动类型推断为不空。不过这里不应该用，我改掉了。

File: components/calendar/__tests__/index.test.js [hunk 2]
```diff
@@ -134,9 +134,9 @@ describe('Calendar', () => {
     const onPanelChangeStub = jest.fn();
     const wrapper = mount(<Calendar mode={yearMode} onPanelChange={onPanelChangeStub} />);
     expect(wrapper.state().mode).toEqual(yearMode);
-    wrapper.instance().setType('date');
+    wrapper.setProps({ mode: monthMode });
     expect(wrapper.state().mode).toEqual(monthMode);
-    expect(onPanelChangeStub).toHaveBeenCalledTimes(1);
+    expect(onPanelChangeStub).toHaveBeenCalledTimes(0);
```
Comment (afc163) [inline components/calendar/__tests__/index.test.js]: 这个用例原来的逻辑极其诡异，看了半天没看懂是啥意思。

**Comment by afc163** [comment]
This pull request **fixes 1 alert** when merging bd0eb344876e07ed913f2138fa438ec7a5d6c076 into 120ea3a38d356f36fa3643e768332936b8be04ec - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-733a4dcb1251f5034c1d650b7e67a50da83212db)

**fixed alerts:**

* 1 for Potentially inconsistent state update

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: components/transfer/index.tsx [hunk 3]
```diff
@@ -192,12 +156,10 @@ class Transfer extends React.Component<TransferProps, any> {
       }
     });
 
-    this.separatedDataSource = {
```
Comment (afc163) [inline components/transfer/index.tsx]: 这个会引发 Trensfer 失效，先删了，本身也不是好的优化。

**Pull request merged**